### ISSUE TITLE
private-call linter should be renamed to private-var-usage

### ISSUE THREAD

**Issue by borkdude** [clj-kondo/clj-kondo#692]
- [ ] To preserve backwards compatibility we will keep supporting the name `:private-call`.
- [ ]  We should also write a unit test for the following:
  - Without the namespace local config it gives two private usages warnings in the foo namespace.
  - With the namespace local config it doesn't give any warnings.
- [ ] document how to use this in config.md. 
- [ ] also document the `#'foo/private-var` workaround 

``` clojure
(ns bar)

(defn- bar [])
(bar)

(ns foo
  {:clj-kondo/config '{:linters {:private-call {:level :off}}}}
  (:require [bar :refer [bar]]))

(bar)
(map bar [])
```
